src/main/java/emu/grasscutter/utils/Utils.java

```diff
@@ -200,8 +201,10 @@ public static void startupCheck() {
 			createFolder(dataFolder);
 
 		// Make sure the data folder is populated, if there are any missing files copy them from resources
-		DataLoader.CheckAllFiles();
-
+		if(Grasscutter.class.getResource("").getProtocol() == "jar"){
```

Comment:
Could you explain this change?

Reply:
DataLoader.CheckAllFiles() only works when running in jar (grasscutter-xxx-dev.jar),

Otherwise it will throw out an error:
java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "filenames" is null
	at emu.grasscutter.data.DataLoader.CheckAllFiles(DataLoader.java:58)
	at emu.grasscutter.utils.Utils.startupCheck(Utils.java:208)
	at emu.grasscutter.Grasscutter.<clinit>(Grasscutter.java:77)

Reply:
I think this needs more attention and be fixed properly in separate PR. Your fix will just ensure it won't crash but it won't copy necessary files. Also the log message is pointless.

Reply:
I know, but I think those who uses ide to run grasscutter know how to copy the data from resources to /data